### PR TITLE
Render Drawings using the Image control and ImageBrush [WIP]

### PR DESCRIPTION
**Pull request by aelij**
* Base interface `IImage` for `IBitmap` and `IDrawing`
* Remove `DrawingPresenter` and unify with the `Image` control
* DX, Cairo & Skia changes to draw a `Drawing` as a brush (Skia is not working properly, please review)

### CODE REVIEW

File: samples/RenderTest/RenderTest.csproj [hunk 1]
```diff
@@ -186,6 +186,11 @@
       <SubType>Designer</SubType>
     </EmbeddedResource>
   </ItemGroup>
+  <ItemGroup>
+    <PackageReference Include="SkiaSharp">
+      <Version>1.57.1</Version>
```
Comment (kekekeks) [inline samples/RenderTest/RenderTest.csproj]: Don't reference packages directly, import the corresponding `props` file from `build` directory

File: src/Avalonia.Visuals/Rendering/DeferredRenderer.cs [hunk 2]
```diff
@@ -414,7 +415,11 @@ private void OnRenderLoopTick(object sender, EventArgs e)
                     Interlocked.Exchange(ref scene, _scene);
                     Render(scene);
                 }
-                catch { }
+                catch (Exception ex)
+                {
+                    Debug.Fail("Fail during render: " + ex.Message, ex.ToString());
+                    throw;
```
Comment (kekekeks) [inline src/Avalonia.Visuals/Rendering/DeferredRenderer.cs]: You are changing behavior from swallowing exception silently to crashing the app without any way to recover.

**Review by kekekeks — CHANGES_REQUESTED** [review]
1) This PR contains tons of formatting changes unrelated to the new functionality. Please, don't do that

2) I don't like the fact that you implement Drawing -> Bitmap conversion for each of drawing backends and essentially copy-pasting some of conversion code from visual brush rendering implementation. Conversion from drawing to bitmap should be shared across drawing backends and visual/drawing brushes should be handled by the same code. We'll have unnecessary code duplication otherwise

File: src/Avalonia.Visuals/Rendering/DeferredRenderer.cs (hunk 2 above)
Comment (aelij) [reply src/Avalonia.Visuals/Rendering/DeferredRenderer.cs]: I discussed this with @grokys - exceptions there caused severe unrecoverable rendering issues. IMO the app should crash in those cases.

**Comment by aelij** [comment]
@kekekeks

1. Can you point me to formatting changes? I was careful not to make those. Are you referring to the usage of pattern matching I introduced (`if (x is Y y)`)?

2. Do you mean this piece of code?

```csharp
using (var drawingContext = new DrawingContext(context))
{
    drawingContext.PushClip(calc.IntermediateClip);
    drawingContext.PushPreTransform(calc.IntermediateTransform);
    drawing.Draw(drawingContext);
}
````

**Comment by kekekeks** [comment]
1) Yep, I'm referring to pattern-matching related changes. They do make code cleaner and easier to read, but IMO we shouldn't include such changes in PRs that are adding new functionality, since  they make changes in untouched blocks of code like that: https://github.com/AvaloniaUI/Avalonia/pull/1149/files#diff-70b4ab66ca4cbc93dbeb866b9cfaab43L185

That makes it harder to reason about actual changes that are introduced by PR. It's not a merge blocker, just refrain from doing that in the future.

2) Yep. This logic - https://github.com/aelij/Avalonia/blob/63f2cececa99769294514d02045a2759a4c3ee5a/src/Skia/Avalonia.Skia/DrawingContextImpl.cs#L233 could be shared across drawing backends. And it also duplicates logic for handling IVisualBrush there - https://github.com/aelij/Avalonia/blob/63f2cececa99769294514d02045a2759a4c3ee5a/src/Skia/Avalonia.Skia/DrawingContextImpl.cs#L210

So we should introduce `IIntermediateBrushRenderer` (instead of VisualBrushRenderer) that encapsulates visual/drawing->bitmap conversion and make backends ignorant about the nature of the brush, be it visual or drawing.

**Comment by jmacato** [comment]
Closing this due to inactivity and the merge conflict will be unnervingly painful to fix

**Pull request closed without merging**